examples/all-clusters-app/all-clusters-common/src/soil-measurement-stub.cpp

```diff
@@ -36,30 +40,58 @@ static const Globals::Structs::MeasurementAccuracyStruct::Type kDefaultSoilMoist
         kDefaultSoilMoistureMeasurementLimitsAccuracyRange)
 };
 
-namespace {
-static std::unique_ptr<Instance> gSoilMeasurementInstance;
-} // namespace
+LazyRegisteredServerCluster<SoilMeasurementCluster> gServer;
+
+constexpr EndpointId kEndpointWithSoilMeasurement = 1;
 
-Instance * SoilMeasurement::GetInstance()
+bool ValidEndpointForSoilMeasurement(EndpointId endpoint)
 {
-    return gSoilMeasurementInstance.get();
+    if (endpoint != kEndpointWithSoilMeasurement)
+    {
+        ChipLogError(AppServer, "SoilMeasurement cluster invalid endpoint");
+        return false;
+    }
+    return true;
 }
 
-void SoilMeasurement::Shutdown()
+} // namespace
+
+void emberAfSoilMeasurementClusterServerInitCallback(EndpointId endpoint) {}
```

Comment:
This function seems to be unused, is it possible to remove it from the generated code?

Reply:
I believe we have __weak__ linkage stubs automatically, so we can remove these.

We cannot remove all and that is a sideffect of ember. So remove and if it fails to link great (I think this can be removed). If others cannot be removed, keep them empty with a comment of "ember framework & code generation requires these to exist".